public/js/utils.js

```diff
@@ -0,0 +1,90 @@
+async function enableEthereum () {
+  if (window.ethereum) {
+    console.log('New Ethereum API detected!')
+    metamask = new Web3(window.ethereum)
+    try {
+      await window.ethereum.enable()
+      console.log('Ethereum enabled...')
+    } catch (error) {
+      console.error(error)
```

Comment:
Should do an `alert` here to display the message. When I reject in Metamask it sends back: User rejected provider access.

Reply:
Also in case of error should set `metamask` to `undefined` so that `submitTransaction` properly reports missing acocunts.

Reply:
done